clarification quota with build/deployment configs

gabemontero opened the pull request:
@derekwaynecarr FYI - I added a section to the quota doc for helping newbies bring up quota for the first time (at least I hit this hiccup).

I didn't see anything equivalent in either the cluster admin/overcommit section, or the resource limits and compute resources sections of the dev guide (granted I skimmed more than read line for line).

I did toy with putting in additional information/examples around limits, but in the end, decided to keep it as simple as possible.

Anyway, PTAL when you get the chance and let me know if you are amenable to such a change (perhaps with some edits) and I'll ping to doc folks to edit/merge, or if you think this is superfluous / covered elsewhere, in which case I'll just nuke the PR / branch.

thanks

derekwaynecarr commented:
I would prefer we add this documentation to the builds and deployments pages.

So a section in each that says, "If you are running with quota, you must set an explicit resource request, here is an example.  If no request is specified, and your project has a `LimitRange` (see https://docs.openshift.org/latest/dev_guide/limits.html), then any defaults defined in that `LimitRange` will apply to pods created during the build/deployment process".

WDYT?

derekwaynecarr commented:
So maybe clean up how we describe resources here:
https://docs.openshift.org/latest/dev_guide/deployments.html#deployment-resources

And do something similar for builds?

gabemontero commented:
That works for me.  I'll circle back (may be a day or two) and rework accordingly.

Would you be amenable if, while removing all the detail from the quota page, I add a 1 sentence reference (with link pointing to) to the deploy and build config pages?

Thanks.

gabemontero commented:
Ok, I moved things (along with some reduction) to the deployment page, leaving a link on the quota page (for now pending your feelings on it.

Also, I remembered that it is still not an apples to oranges compare wrt build, in particular their honoring the limits, and that @bparees is working on the card to address this sprint, and that doc updates would be arising from that.  So I removed the BuildConfig piece for now (even though the requests pieced is still needed), assuming that it would be filled back in  along with the other details resulting from @bparees 's work, and figured whatever placeholder I added would be throwaway.

gabemontero commented:
hey @derekwaynecarr , as part of this https://github.com/openshift/origin/pull/6858, Clayton informed me about changes happening with "project quota defaults", where we wouldn't require the presents of a resources->request or limit if a quota has been placed on the project.

If I understood that point correctly, that pretty much renders this doc tweak null and void, no?  If so, I'll just nuke it.

thanks

derekwaynecarr commented on dev_guide/deployments.adoc:
```diff
@@ -653,6 +653,32 @@ In the following example, each of `resources`, `cpu`, and `memory` is optional.
+* your project also has a link:./limits.html[LimitRange] defined, where the defaults from the `LimitRange` will apply to pods
+created during the deployment process
+
+Otherwise, deploy pod creation will fail, citing the need to `make a non-zero request for memory since it is tracked by quota`.
```
The error string is going to change shortly upstream, I would prefer we just say that the pod creation will fail citing a failure to satisfy quota.

gabemontero replied:
thx - part of next push

derekwaynecarr commented:
@gabemontero - sorry, this fell off my radar.  Just tweak the part where you appear to include the actual error string to be more generic since that string will change soon in upstream Kube.

adellape commented on dev_guide/deployments.adoc:
```diff
@@ -653,6 +653,32 @@ In the following example, each of `resources`, `cpu`, and `memory` is optional.
+    }
+  },
+}
+----
```
Convert to YAML:

```
[source,yaml]
----
  type: "Recreate"
  resources: 
    requests: <1>
      cpu: "100m"
      memory: "256Mi"
----
```

gabemontero replied:
part of next push

adellape commented on dev_guide/quota.adoc:
```diff
@@ -243,3 +243,9 @@ The `*resource-quota-sync-period*` setting is designed to balance system
 the master.
 ====
+
+[[accounting_quota_dc]]
```
```
[[accounting-quota-dc]]
```

gabemontero replied:
changed to hyphens ... part of next push ... though I'm mention the other section heads appear to use underscores

adellape commented on dev_guide/quota.adoc:
```diff
@@ -243,3 +243,9 @@ The `*resource-quota-sync-period*` setting is designed to balance system
+
+[[accounting_quota_dc]]
+
+== Accounting for Quota in Deployment Configs
```
s/Configs/Configurations/

gabemontero replied:
part of next push

adellape commented on dev_guide/quota.adoc:
```diff
@@ -243,3 +243,9 @@ The `*resource-quota-sync-period*` setting is designed to balance system
+
+== Accounting for Quota in Deployment Configs
+
+If a quota has been defined for your project, read about the considerations for any Deployment Configs link:./deployments.html#deployment-resources[here].
```
If a quota has been defined for your project, see link:../dev_guide/deployments.html#deployment-resources[Deployment Resources] for considerations on any deployment configurations.

gabemontero replied:
thx ... part of next push

gabemontero commented:
hey @derekwaynecarr @adellape  - I've pushed the updates you all asked for.  Thanks.

gabemontero commented:
hey @derekwaynecarr @adellape - any chance you all will have the bandwidth to see my updates based on the last set of comment, and possibly OK this change?

adellape commented:
@gabemontero LGTM

derekwaynecarr commented:
LGTM

adellape commented:
:shipit:

adellape commented:
revhistory in https://github.com/openshift/openshift-docs/pull/1885
